tensordict/prototype/tensorclass.py

```diff
@@ -166,7 +173,11 @@ def wrapper(self, *args, batch_size=None, device=None, _tensordict=None, **kwarg
                     f"Keys from the tensordict ({set(_tensordict.keys())}) must "
                     f"correspond to the class attributes ({expected_keys})."
                 )
-            input_dict = {key: None for key in _tensordict.keys()}
+            # Handling Non-Tensor data
+            if _non_tensordict:
```

Comment:
I prefer the more explicit `_non_tensordict is not None` otherwise there's a risk that someone inputs an empty dict or something like it and hopes that it will return True when it won't.

Reply:
In the function _get_non_tensor_dict in the line 549, I had to initialize an empty dict and add key value pairs if the value is a non-tensor value and hence used without None

Reply:
so what are we testing precisely? That it is empty? 
In fact I know it's a very pythonish way of doing things (and very common) but IMO it doesn't make the code very readable. `len(dictionary) == 0` or something similar is more verbose but helps understand what we're testing.

Reply:
Sure. Makes sense. Will update the code to add the explicit condition.

Reply:
Not relevant any more